app/code/core/Mage/Catalog/Model/Resource/Eav/Attribute.php

```diff
@@ -288,6 +287,15 @@ public function getFrontendLabel()
         return $this->_getData('frontend_label');
     }
 
+    /**
+     * Retrieve is_filterable value
+     * @return int
+     */
+    public function getIsFilterable()
```

Comment:
What is the point of making this a real method, it appears to be functionally equivalent?

Reply:
In a word: performance. According to xdebug profiling it is often called several thousands of times on catalog pages when you have a decent amount of filter options. __call is fairly expensive in comparison and it saved ~1.5-2ms on our production store.

Reply:
Not sure where we draw the line on that but maybe just add a note to the vardoc then like "Optimization to avoid __call".

An interesting idea would be a script that replaces all @method annotations with functions at build time.. A thousand micro-optimizations makes at least a milli-optimization, right?

Reply:
In general, yes the performance gain is marginal for most cases but this one accounted for more than half the __call invocations on pages with product lists according to xdebug profiler.

Generating all phpdoc methods actually is an interesting idea, but won't be enough. There is a ton of fields around the code that are not documented. Maybe I'll look into it.